yt_dlp/extractor/murrtube.py

```diff
@@ -0,0 +1,168 @@
+    viewsCount
+    thumbnailKey
+    tagList
+    user {
+      name
+      __typename
+    }
+    __typename
+  }
+}'''})
+        meta = data['medium']
+
+        storage_url = 'https://storage.murrtube.net/murrtube/'
+        format_url = storage_url + meta.get('key', '')
+        thumbnail = storage_url + meta.get('thumbnailKey', '')
+
+        formats = []
+        if determine_ext(format_url) == 'm3u8':
+            formats.extend(self._extract_m3u8_formats(
+                format_url, video_id, 'mp4', entry_protocol='m3u8_native',
+                fatal=False))
+        else:
+            formats.append({
+                'url': format_url,
+            })
+
+        return {
+            'id': video_id,
+            'title': meta.get('title'),
+            'description': meta.get('description'),
+            'formats': formats,
+            'thumbnail': thumbnail,
+            'duration': int(meta.get('duration', 0)),
+            'uploader': meta.get('user', {}).get('name'),
+            'view_count': meta.get('viewsCount'),
+            'like_count': meta.get('likesCount'),
+            'comment_count': meta.get('commentsCount'),
+            'tags': meta.get('tagList'),
+            'age_limit': 18,
+        }
+
+
+class MurrtubeUserIE(MurrtubeIE):
+    IE_DESC = 'Murrtube user profile'
+    _VALID_URL = r'https?://murrtube\.net/(?P<id>[^/]+)$'
+    _TEST = {
+        'url': 'https://murrtube.net/stormy',
+        'info_dict': {
+            'id': 'stormy',
+        },
+        'playlist_mincount': 27,
+    }
+    _PAGE_SIZE = 10
+
+    def _fetch_page(self, username, user_id, page):
+        data = self._download_gql(username, {
+            'operationName': 'Media',
+            'variables': {
+                'limit': self._PAGE_SIZE,
+                'offset': page * self._PAGE_SIZE,
+                'sort': 'latest',
+                'userId': user_id,
+            },
+            'query': '''query Media($q: String, $sort: String, $userId: ID, $offset: Int!, $limit: Int!) {
+media(q: $q, sort: $sort, userId: $userId, offset: $offset, limit: $limit) {
+id
+__typename
+}
+}'''},
+            'Downloading page {0}'.format(page + 1))
+        if data is None:
+            raise ExtractorError(f'Failed to retrieve video list for page {page + 1}', expected=True)
+
+        media = data['media']
+
+        for entry in media:
+            yield self.url_result(
+                'murrtube:{0}'.format(entry['id']),
+                MurrtubeIE.ie_key())
+
+    def _real_extract(self, url):
+        username = self._match_id(url)
+        data = self._download_gql(username, {
+            'operationName': 'User',
+            'variables': {
+                'id': username,
+            },
+            'query': '''query User($id: ID!) {
+  user(id: $id) {
+    id
+    __typename
+  }
+}'''},
+            'Downloading user info')
+        if data is None:
+            raise ExtractorError('Failed to fetch user info', expected=True)
```

Comment:
why expected?

Reply:
This is the error you receive when a username is invalid, so my assumption was that this would be the most likely situation where this error would be displayed. It's probably better to also handle when the site changes, so thinking about it now I will remove.